### ISSUE TITLE
3.4+ error can't get just created node by id

### ISSUE THREAD

**Issue by peterfarabaugh** [neo4j/neo4j#12144]
EDIT: removing the disableLosslessIntegers from the driver initialization stops this issue, but then any integer data I retrieve comes back as {low: 10, high: 0} type structures which breaks my entire application.

this issue exists on 3.4 and 3.51 (maybe other post 3.3). I am running on a AWS Ubuntu with neo4j bundled. I am using 1.7.2 of the neo4j-driver for node (although 1.7.0 failes the same way). The client side is running on windows 10

The following code works fine on neo4j 3.3 using the 1.7.2 neo4j-driver for node:
```
import {v1 as neo4j} from 'neo4j-driver';

const url: string = process.env.COREDB_URL || '';
const user: string = process.env.COREDB_USERNAME || '';
const password: string = process.env.COREDB_PASSWORD || '';

const driver = neo4j.driver(url, neo4j.auth.basic(user, password), {disableLosslessIntegers: true});
let connection = driver.session()


async function go() {
    let res = await connection.run(`create (b:Banana {tag: 'test'}) return b,id(b) as id`, {});
    let b = res.records[0].get('b').properties
    console.log('b',b)
    let id = res.records[0].get('id')
    console.log('id',id)
    res = await connection.run(`MATCH (u) where id(u)=$id return u as id`, {id: id});
    console.log(res.records)
    let id2 = res.records[0].get('id').properties;
    console.log('id2',id2)
}

go().then(() => console.log('done')).catch((e) => console.log(e.message))
```
it gives the following output:
```
> node tools\test-id.js
b { tag: 'test' }
id 1858404
[ Record {
    keys: [ 'id' ],
    length: 1,
    _fields: [ [Node] ],
    _fieldLookup: { id: 0 } } ]
id2 { tag: 'test' }
done
```
Under 3.5.1 it does not work. The second statement returns no records:
```
> node tools\test-id.js
b { tag: 'test' }
id 1856012
[]
Cannot read property 'get' of undefined
```
BTW, the reason I need to do the get by id right after the create is that I am using an apoc trigger to add things to the node after creation, and apoc triggers apparently run after the object is created and returned, so I need the second get to see the transformed node.
But, for this distilled example I removed the trigger from my DB to ensure that it was not causing the issue

**Comment by ali-ince** [comment]
Hi @peterfarabaugh,

Thanks for reporting the issue. The main thing that causes this behaviour change is, when `disableLosslessIntegers` is `false` (default), neo4j integer values are correctly typed as our driver level integer class - which holds the symmetry property between the client and the server (i.e. when you do a `RETURN $x` with `x` set as a value of type `neo4j.types.Integer`, you will receive back the same value with the same type). However, when `disableLosslessIntegers` is `true`, this no longer holds - because what we get from the server is always deserialized as javascript numbers and javascript numbers are serialized as double values while sending them to the server.

So, in one query (when `disableLosslessIntegers: false`), you send to the server something like `MATCH (n) WHERE id(n) = 20 RETURN n` - whereas in the other case it becomes  `MATCH (n) WHERE id(n) = 20.0 RETURN n`. 

It seems that there's a bug around handling this two variants. However, for a workaround, you could pass the `id` parameter as `Integer` type by passing it through `neo4j.int(id)` function. Like;

```
res = await connection.run(`MATCH (u) where id(u)=$id return u as id`, {id: neo4j.int(id)});
```

I'm assigning the issue to the relevant team, and the issue will be resolved when an actual fix is in place to get rid of this difference.

Thanks.

**Comment by sherfert** [comment]
This will be fixed in the next 3.4 patch release.
